### PR TITLE
Add active link classes to site navigation menu

### LINKED ISSUE #178: Missing active class on the header links
I want to highlight the header section which is currently active but there's no indication right now.

Example on the RN website:
image

### PR DESCRIPTION
**Pull request by microbouji**
## Motivation

Fixes #178 

Read the doc change at guides-navigation.md to see how it works.

For the main styles ( and thus the docusaurus site) I only added the active classes to the already existing hover and focus rules.

## Test Plan

I tested docs with two sidebars, blog posts, and `.js` and `.html` user pages with `wrapPagesHTML`.
For testing versioned and translated docs I manually copied some files instead of setting up crowdin.

## Related PRs

I read this part of the new PR template after having already included the doc changes in this one :smile:. Let me know if I should move them to a separate PR.

### CODE REVIEW

**Comment by JoelMarcey** [comment]
So nice! 👍

File: lib/core/nav/HeaderNav.js [hunk 1]
```diff
@@ -172,8 +179,16 @@ class HeaderNav extends React.Component {
       // set link to blog url
       href = this.props.baseUrl + 'blog';
     }
+    const itemClasses = classNames({
+      siteNavGroupActive:
+        (link.doc && docGroupActive) || (link.blog && this.props.current.blog),
+      siteNavItemActive:
+        docItemActive ||
+        (link.blog && this.props.current.blogListing) ||
```
Comment (vjeux) [inline lib/core/nav/HeaderNav.js]: I'm not sure what's going on with the blog but it doesn't seem to be highlighted when I'm looking at the blog section. All the other sections are working fine (thanks for fixing this!!)

Comment (vjeux) [reply lib/core/nav/HeaderNav.js]: Nevermind, I actually needed to change the headerLink from

```js
    { href: "/blog", label: "Blog" },
```

to

```js
    { blog: true, label: "Blog" },
```

now it works :)

Comment (yangshun) [reply lib/core/nav/HeaderNav.js]: Is this a backward-incompatible change? Do we need to document this somewhere or support the previous?

Comment (microbouji) [reply lib/core/nav/HeaderNav.js]: We added the corresponding documentation when adding this feature to explain it doesn't work with "href" header links: https://docusaurus.io/docs/en/navigation.html#active-links-in-site-navigation-bar

Comment (vjeux) [reply lib/core/nav/HeaderNav.js]: It's not really backward incompatible, href: "/blog" still works but doesn't have the highlight (note that it didn't before either!). If you are saying to use blog: true in the docs, this should be fine, I wrote this before this boolean existed.

**Pull request merged**